### ISSUE TITLE
`StreamCombinatorsSuite.evalFilterNotAsync - filters up to N items in parallel` is flaky on Scala.js

### ISSUE THREAD

**Issue by vasilmkd** [typelevel/fs2#2593]
I noticed this while running the tests locally.

```scala
==> X fs2.StreamCombinatorsSuite.evalFilterNotAsync - filters up to N items in parallel 0.91s munit.ComparisonFailException: /Users/vasil/Code/fs2/core/shared/src/test/scala/fs2/StreamCombinatorsSuite.scala:489
488:        }
489:        .assertEquals(n -> 0)
490:    }
values are not the same
=> Obtained
Tuple2(
  _1 = 4,
  _2 = 0
)
=> Diff (- obtained, + expected)
 Tuple2(
-  _1 = 4,
+  _1 = 5,
   _2 = 0
```

**Comment by armanbilge** [comment]
I bisected this to https://github.com/typelevel/cats-effect/pull/2207.

**Comment by armanbilge** [comment]
Proposed a fix in https://github.com/typelevel/fs2/pull/2651#discussion_r741550531.

**Comment by armanbilge** [comment]
This can be closed as of #2561.
